src/main.cpp

```diff
@@ -3979,7 +3979,6 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             mempool.check(pcoinsTip);
             RelayTransaction(tx);
             vWorkQueue.push_back(inv.hash);
-            vEraseQueue.push_back(inv.hash);
```

Comment:
Why can't we mark a tx for removal already? It was accepted into the memory pool, so alls its inputs must be satisfied?